crypto/ring-signature/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! MobileCoin implementation of amount commitments and MLSAG ring signatures,
+//! as well as some related functions (see one-time keys module)
+
+#![no_std]
+#![deny(missing_docs)]
+
+extern crate alloc;
+
+use crate::onetime_keys::create_shared_secret;
+use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
+
+mod amount;
+mod domain_separators;
+mod ring_signature;
+
+pub mod onetime_keys;
+#[cfg(any(test, feature = "proptest"))]
+pub mod proptest_fixtures;
+
+pub use amount::{Commitment, CompressedCommitment};
+pub use ring_signature::{
+    generators, CryptoRngCore, CurveScalar, Error, GeneratorCache, KeyImage, PedersenGens,
+    ReducedTxOut, RingMLSAG, Scalar,
+};
+
+/// Get the shared secret for a transaction output.
+///
+/// # Arguments
+/// * `view_key` - The recipient's private View key.
+/// * `tx_public_key` - The public key of the transaction.
+pub fn get_tx_out_shared_secret(
```

Comment:
I think it was mainly for documentation, because the variable names tell you what key you are supposed to combo with what

I don't think we should delete this in this PR, it's disruptive and not in scope of what we were trying to do

Reply:
I guess I think what we should actually do is delete `create_shared_secret`